connectors/src/lib/dust_api.ts

```diff
@@ -381,22 +381,49 @@ export class DustAPI {
     return new Ok(json.conversation as ConversationType);
   }
 
-  async streamAgentMessageEvents({
-    conversation,
+  async postUserMessage({
+    conversationId,
     message,
   }: {
-    conversation: ConversationType;
-    message: AgentMessageType;
+    conversationId: string;
+    message: PostMessagesRequestBodySchema;
+  }) {
+    const res = await fetch(
+      `${DUST_API}/api/v1/w/${this.workspaceId()}/assistant/conversations/${conversationId}/messages`,
+      {
+        method: "POST",
+        headers: {
+          Authorization: `Bearer ${this._credentials.apiKey}`,
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          ...message,
+        }),
+      }
+    );
+
+    const json = await res.json();
+    if (json.error) {
+      return new Err(json.error as DustAPIErrorResponse);
+    }
+
+    return new Ok(json.message as UserMessageType);
+  }
+
+  async streamAgentMessageEvents({
+    conversationId,
```

Comment:
why change the signature here?

Reply:
I initially thought I was going to have a code path where I had the message id but not the agentMessageType itself.
That being said, it's an API client and the API only needs the id, so having the ConversationType and AgentMessageType make using this API a bit harder.

Reply:
More restrictive but cleaner let's not change it if not needed.

Reply:
Sounds good, I will fix it before merging. I am working on content fragment right now.